### PR TITLE
feat: enable jsx transform

### PR DESCRIPTION
**Pull request by herschel666**
<details>
<summary>Bors merge bot cheat sheet</summary>

We are using [bors-ng](https://github.com/bors-ng/bors-ng) to automate merging of our pull requests. The following table provides a summary of commands that are available to reviewers (members of this repository with push access) and delegates (in case of `bors delegate+` or `bors delegate=[list]`).

| Syntax | Description |
| --- | --- |
| bors merge | Run the test suite and push to master if it passes. Short for "reviewed: looks good." |
| bors merge- | Cancel an r+, r=, merge, or merge= |
| bors try | Run the test suite without pushing to master. |
| bors try- | Cancel a try |
| bors delegate+ | Allow the pull request author to merge their changes. |
| bors delegate=[list] | Allow the listed users to r+ this pull request's changes. |
| bors retry | Run the previous command a second time. |

This is a short collection of opinionated commands. For a full list of the commands read the [bors reference](https://bors.tech/documentation/).

</details>

### CODE REVIEW

File: packages/spec/integration/react-16/mixin.core.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { Mixin } = require('hops-mixin');
+
+class CoreMixin extends Mixin {
+  configureBuild(webpackConfig) {
+    webpackConfig.resolve.alias = webpackConfig.resolve.alias || {};
+    webpackConfig.resolve.alias['react$'] = require.resolve('react16.8');
```
Comment (ZauberNerd) [inline packages/spec/integration/react-16/mixin.core.js:6]: Why do we need this?

File: packages/spec/package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
     "esbuild-loader": "^2.10.0",
     "jest-environment-hops": "15.0.0-nightly.1",
     "raw-body": "^2.4.1",
+    "react16.8": "npm:react@16.8.0",
```
Comment (ZauberNerd) [inline packages/spec/package.json:44]: Why do we need to specifically request react@16.8? Shouldn't the react that is already available be sufficient?

Comment (herschel666) [reply packages/spec/package.json:44]: React 16.8 is part of our supported range. But I can as well remove the specific test for it.

Comment (herschel666) [reply packages/spec/package.json:44]: @ZauberNerd How do we want to go on with the additional test for React 16.8?

Comment (ZauberNerd) [reply packages/spec/package.json:44]: I'd say we can remove that, since we have react@17 in our repository anyway.

Comment (herschel666) [reply packages/spec/package.json]: @ZauberNerd PTAL again

**Review by ZauberNerd — APPROVED** [review]

**Comment by herschel666** [comment]
bors merge

**Pull request merged**